### DISCUSSION TITLE
1v1 Versus Mode

### DISCUSSION THREAD

**Discussion by WalmartWhiteCat** [ppy/osu#30540] [Ideas]
I don't exactly know how people would feel about this, but I simply have to say it because the idea is genuinely so cool to me. Like the title implies, it would be a way to pair players up via a queue, and then these players can take turns picking maps and 1v1ing on them until one wins. The main details are this:

- New Multiplayer mode called 1v1 Versus
- New ranking system just for 1v1 Versus
- Rank based matchmaking
- Round Robin map picks
- Geoguessr-style health system (1000 health per player. The better you do the more damage your opponent takes at the end of the map. But the better your opponent does the more they can block and possibly even do damage to you instead. I.e. You do 800dmg but your opponent does 700dmg, they only take 100dmg)
- Possible 2v2 mode if the format is well received

I really hope this is heavily considered, because it seems like something a massive part of the community would absolutely love. I know for a fact I at least would grind the hell out of this mode if it ever came into the game.

(Okay istg I searched for 1v1 and didn't see anything in Discussions please don't yell at me ;-;)

**Comment by valarics** [comment]
thats my goat 🐈

**Comment by Anymusxd** [comment]
we know it's you btmc

**Reply by WalmartWhiteCat** [reply]
nuh uh

**Comment by pigeonskew** [comment]
I don't think letting the players pick the maps is the right choice since players would likely pick maps that they've already practiced or choose mods that the other player can't play good at just to get an easy win.

I think an online system should be implemented that will fetch each player's recommended difficulty, average it, pick a map that matches that difficulty and enable a random mod or not. That way, it's fair for both players.

**Reply by Anymusxd** [reply]
how about map voting
or if they both agree on one that's the played map

**Reply by pigeonskew** [reply]
Even better, maybe just adding a "Free Pick" toggle under the 1v1 versus mode would switch between 1v1 with map picks from players and 1v1 with automated map picks like the online system I described.

**Reply by pigeonskew** [reply]
> how about map voting or if they both agree on one that's the played map

but there's the possibility of the other player not letting you play the maps you pick that you're good at and if both players were smart, they'd just prevent the other player to play any map they pick by voting not to play it lol.

**Reply by WalmartWhiteCat** [reply]
I actually didn’t think about that. Yeah, good point. Probably having a veto system or something would be cool. Like vetoing maps above a certain bpm. Or a certain SR. And voting system is also a fine idea, but ideally it should be seamless (I.e. the players join the lobby and the matches run without any extra input). 

I think utilizing the new tagging system that was implemented would be good. The idea would be for each player’s profile to be scanned at the start of the lobby. During this, a system would look at the top 100 maps on each profile and find 3-5 common tags between the players. Then it finds the average SR between the players’ top 100s and filters map selection based off these parameters.

Tags are a community-based system already, so this should ensure that the filter is as accurate as possible. 

The scan probably shouldn’t run every single time a player joins a Versus lobby. I think the best idea would be to run it only once per week, or even month. This ensures there isn’t too much strain on the servers. Just one scan per week/month, then the player is added to a list of other players that outlines which players can be matched with each other (basically like a sorting system. Players determined to play tech and finger control most often will be added to a list of other players that most accurately fit that playstyle).

**Reply by WalmartWhiteCat** [reply]
And yeah, the system for choosing maps should be automatic. No voting or anything. It adds the maps on its own and applies any mods on its own. Player input should be kept to a minimum, to streamline the process and keep things fair.

Of course, some players won’t like this. So maybe adding a “Freeform” queue would be nice. Maybe you don’t want to challenge yourself very hard, or you just wanna vibe and pick maps you think are fun. Then you join the Freeform lobby and do that.
